contracts/FlowIDTableStaking.cdc

```diff
@@ -249,40 +220,56 @@ pub contract FlowIDTableStaking {
         }
 
         /// borrow a reference to to one of the delegators for a node in the record
-        access(account) fun borrowDelegatorRecord(_ delegatorID: UInt32): &DelegatorRecord {
+        access(account) view fun borrowDelegatorRecord(_ delegatorID: UInt32): auth(FungibleToken.Withdraw) &DelegatorRecord {
             pre {
                 self.delegators[delegatorID] != nil:
                     "Specified delegator ID does not exist in the record"
             }
-            return (&self.delegators[delegatorID] as &DelegatorRecord?)!
+            return (&self.delegators[delegatorID] as auth(FungibleToken.Withdraw) &DelegatorRecord?)!
         }
 
         /// Add a delegator to the node record
         access(account) fun setDelegator(delegatorID: UInt32, delegator: @DelegatorRecord) {
             self.delegators[delegatorID] <-! delegator
         }
+
+        access(account) fun setDelegatorIDCounter(_ newCounter: UInt32) {
+            self.delegatorIDCounter = newCounter
+        }
+
+        access(account) fun setNetworkingAddress(_ newAddress: String) {
+            self.networkingAddress = newAddress
+        }
+
+        access(contract) fun setTokensRequestedToUnstake(_ newUnstakeRequest: UFix64) {
+            self.tokensRequestedToUnstake = newUnstakeRequest
+        }
+
+        access(contract) fun setWeight(_ newWeight: UInt64) {
+            self.initialWeight = newWeight
+        }
     }
 
     /// Struct to create to get read-only info about a node
-    pub struct NodeInfo {
-        pub let id: String
-        pub let role: UInt8
-        pub let networkingAddress: String
-        pub let networkingKey: String
-        pub let stakingKey: String
-        pub let tokensStaked: UFix64
-        pub let tokensCommitted: UFix64
-        pub let tokensUnstaking: UFix64
-        pub let tokensUnstaked: UFix64
-        pub let tokensRewarded: UFix64
+    access(all) struct NodeInfo {
+        access(all) let id: String
+        access(all) let role: UInt8
+        access(all) let networkingAddress: String
+        access(all) let networkingKey: String
+        access(all) let stakingKey: String
+        access(all) let tokensStaked: UFix64
+        access(all) let tokensCommitted: UFix64
+        access(all) let tokensUnstaking: UFix64
+        access(all) let tokensUnstaked: UFix64
+        access(all) let tokensRewarded: UFix64
 
         /// list of delegator IDs for this node operator
-        pub let delegators: [UInt32]
-        pub let delegatorIDCounter: UInt32
-        pub let tokensRequestedToUnstake: UFix64
-        pub let initialWeight: UInt64
+        access(all) let delegators: &[UInt32]
```

Comment:
Doesn't seem to be a problem since the core contracts have been migrated multiple times, but I'm surprised this is a compatible migration. A reference field would mean that the struct isn't storable, wouldn't it?

Reply:
nobody stores this struct so it isn't a problem. It just gives a snapshot of a node's stake at a specific point in time so storing it wouldn't make sense.